extensions/wikia/AbPerfTesting/classes/Hooks.class.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Wikia\AbPerfTesting;
+
+use \Wikia\Logger\WikiaLogger;
+
+class Hooks {
+
+	/**
+	 * Initialize the experiment and set all required tracking things
+	 *
+	 * @param string $experimentName
+	 * @param array $experimentConfig
+	 */
+	private static function startExperiment( $experimentName, Array $experimentConfig ) {
+		wfDebug( sprintf( "%s[%s] using %s class with %s params\n",
+			__METHOD__, $experimentName, $experimentConfig['handler'], json_encode( $experimentConfig['params'] ) ) );
+
+		new $experimentConfig['handler']( $experimentConfig['params'] ? : [] );
+
+		// mark a transaction with an experiment name
+		\Transaction::getInstance()->set( \Transaction::PARAM_AB_PERFORMANCE_TEST, $experimentName );
+
+		// mark a transaction using UA's custom dimensions
+		global $wgHooks;
+		$wgHooks['WikiaSkinTopScripts'][] = function( Array &$vars, &$scripts ) use ( $experimentName ) {
+			$val = \Xml::encodeJsVar( $experimentName );
+			$scripts .= \Html::inlineScript( "_gaq.push(['set', 'dimension20', {$val}]);" );
+			return true;
+		} ;
+
+		/*
+		 * Start the session to bypass CDN cache
+		 *
+		 * We don't want to polute the CDN cache with the A/B performance testing tracking data.
+		 * As the test are run for only a small subset of the traffic, start the session for client
+		 * that are in the test groups to bypass the CDN cache.
+		 */
+		if ( session_id() == '' ) {
+			wfSetupSession();
+			wfDebug( __METHOD__ . " - session started\n" );
+
+			// log started sessions
+			global $wgUser;
+			WikiaLogger::instance()->info( __METHOD__, [
+				'experiment' => $experimentName,
+				'session_id' => session_id(),
+				'is_anon' => $wgUser->isAnon()
+			] );
+		}
```

Comment:
It will, both types of tests (for front-end and back-end) will need a request that hits the backend to be triggered. Given a small articles CDN hit ratio (a bit more than 10%) I think we can live with that in MVP phase of the framework.